### PR TITLE
Fix checkstyle check

### PR DESCRIPTION
**Pull request by im-konge**
### Type of change

- Bugfix

### Description

This PR fixes the checkstyle check after the #10527 - the `[]` brackets are not recognizable by the checkstyle plugin, which caused that all the errors were skipped (mainly in the tests modules folders/modules). That means -> if we remove some class and the import is unused, normally it would cause that we get the `UnusedImport` error from checkstyle. But because all of the errors were skipped, we got the successful build even when the unused import was there. 

Also, some of the checks were wrongly named.

I'm changing the `[]` brackets to `()`, changing the check types to correct ones and also adding more checks that are hit during the build.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Review by jankalinic — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Review by see-quick — APPROVED** [review]

**Review by henryZrncik — APPROVED** [review]

**Review by scholzj — COMMENTED** [review]
Please explain what exactly is the problem because it clearly is passing on the CI.

**Review by scholzj — COMMENTED** [review]
One more comment -> leaving aside that nothing seems to be failing ... you should not exclude all the checks. If you do Javadocs in any of these parts we don't check, you should still do them properly. They should just not be mandatory.

**Comment by im-konge** [comment]
> Please explain what exactly is the problem because it clearly is passing on the CI.

I'm mentioning it in the description - because of the `[]` brackets, all errors (like unused imports) were ignored. It's passing in the CI, because it skips all the errors. Not just those which are written in the suppressions.xml

>  you should not exclude all the checks. If you do Javadocs in any of these parts we don't check, you should still do them properly. They should just not be mandatory.

Sorry, I'm not sure if I understand. I'm just skipping those errors that are related to Javadoc (and which you added in the previous PR). Those skips are related to the tests modules/folders.

**Comment by scholzj** [comment]
> > Please explain what exactly is the problem because it clearly is passing on the CI.
> 
> I'm mentioning it in the description - because of the `[]` brackets, all errors (like unused imports) were ignored. It's passing in the CI, because it skips all the errors. Not just those which are written in the suppressions.xml. That means - if we remove something and we have unused import, the checkstyle will show that the build was successful. Which is wrong.

Ok, I guess that is my mistake. But that is not what I understood from the description so maybe you should make it more clear.

> > you should not exclude all the checks. If you do Javadocs in any of these parts we don't check, you should still do them properly. They should just not be mandatory.
> 
> Sorry, I'm not sure if I understand. I'm just skipping those errors that are related to Javadoc (and which you added in the previous PR). Those skips are related to the tests modules/folders.

The point is that if you use Javadocs, you should still do them correctly. E.g. with the right style, with all prarameters etc. So you should not exclude all the checks. You should only exclude the checks that require the Javadocs for the public fields. I.e., if you don't use Javadocs, you should be fine. If you use them, use them correctly.

**Comment by im-konge** [comment]
> Ok, I guess that is my mistake, sorry. But that is not what I understood from the description so maybe you should make it more clear.

Yeah sorry, I will make it clear in the description.

> The point is that if you use Javadocs, you should still do them correctly. E.g. with the right style, with all prarameters etc. So you should not exclude all the checks. You should only exclude the checks that require the Javadocs for the public fields. I.e., if you don't use Javadocs, you should be fine. If you use them, use them correctly.

Okay, I will check that. Thanks

**Comment by im-konge** [comment]
@scholzj I've updated the checks (removed the skipped error types that I added) and the code based on the checkstyle errors I got. The description is updated as well. Could you please have another look? Thanks!

File: topic-operator/src/test/java/io/strimzi/operator/topic/LogCaptor.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ private LogCaptor(AbstractLogger logger,
      * @param predicate A predicate (e.g. on the message text)
      * @param timeout A timeout
      * @param unit A timeout unit
-     * @return
+     * @return created LogCaptor
```
Comment (scholzj) [inline topic-operator/src/test/java/io/strimzi/operator/topic/LogCaptor.java:75]: You should start with a capital letter. Same below as well ...

```suggestion
     * @return Created LogCaptor
```

File: test/src/main/java/io/strimzi/test/executor/Exec.java [hunk 2]
```diff
@@ -240,9 +240,9 @@ public static ExecResult exec(String input, List<String> command, int timeout, L
      * @param commands arguments for command
      * @param timeoutMs  timeout in ms for kill
      * @return returns ecode of execution
-     * @throws IOException
-     * @throws InterruptedException
-     * @throws ExecutionException
+     * @throws IOException when writing/closing the OutputStreams or starting the process
+     * @throws InterruptedException in the waitFor method, if the wait is interrupted
+     * @throws ExecutionException when getting the output from std
```
Comment (scholzj) [inline test/src/main/java/io/strimzi/test/executor/Exec.java:245]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/MinioUtils.java [hunk 3]
```diff
@@ -24,8 +24,8 @@ private MinioUtils() {
 
     /**
      * Collect data from Minio about usage of a specific bucket
-     * @param namespace
-     * @param bucketName
+     * @param namespace     name of the Namespace where the Minio Pod is running
+     * @param bucketName    name of the bucket for which we want to get info about its size
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/MinioUtils.java:28]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/BuildUtils.java [hunk 4]
```diff
@@ -19,9 +19,9 @@ private BuildUtils() { }
 
     /**
      * Gets OpenShift build name based on name and version
-     * @param name
-     * @param version
-     * @return
+     * @param name      name of the build
+     * @param version   version of the build
+     * @return  returns `name-version` build name
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/BuildUtils.java:24]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java [hunk 5]
```diff
@@ -176,7 +176,7 @@ public static void buildImageOpenshift(String name, String namespace, String doc
      * @param namespace location of the config map
      * @param configMapName name of the config map
      * @param dockerfilePath path to the Dockerfile
-     * @throws IOException
+     * @throws IOException  when reading from file - when file doesn't exist or cannot be opened
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:179]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java [hunk 6]
```diff
@@ -119,7 +119,7 @@ public static void buildImageKaniko(String name, String namespace, String docker
      * @param dockerfilePath path to Dockerfile
      * @param imageTag tag of the final image that will be pushed into internal registries
      * @param baseImage used base image for the build
-     * @throws IOException
+     * @throws IOException  when reading from file - when file doesn't exist or cannot be opened
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:122]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java [hunk 7]
```diff
@@ -61,7 +61,7 @@ public static void buildImage(String name, String namespace, String dockerfilePa
...
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:64]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java [hunk 8]
```diff
@@ -44,7 +44,7 @@ public class ImageBuild {
...
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:47]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/NodePoolsConverter.java [hunk 9]
```diff
@@ -51,7 +51,7 @@ public static KafkaNodePool[] convertNodePoolsIfNeeded(KafkaNodePool... nodePool
      * it does following:
      *  - removes all controller NodePools (so we have just one NodePool instead of two - easier handling in STs)
      *  - for each NodePool that left (broker NodePools) adds {@link ProcessRoles#CONTROLLER} role to its `spec`
-     * @param nodePools
+     * @param nodePools     list of NodePools that should be converted to mixed roles
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/NodePoolsConverter.java:54]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/NamespaceManager.java [hunk 10]
```diff
@@ -198,8 +198,8 @@ public void deleteNamespaceWithWait(String namespaceName) {
      * Deletes Namespace with {@param namespaceName}, waits for its deletion, and in case that {@param collectorElement}
      * is not {@code null}, removes the Namespace from the {@link #MAP_WITH_SUITE_NAMESPACES}.
      *
-     * @param namespaceName
-     * @param collectorElement
+     * @param namespaceName     name of the Namespace that should be deleted
+     * @param collectorElement  collector element for removing the Namespace from the set
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/NamespaceManager.java:202]: You should start with a capital letter.

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/clientproperties/AbstractKafkaClientProperties.java [hunk 11]
```diff
@@ -256,11 +256,10 @@ protected AbstractKafkaClientProperties(KafkaClientPropertiesBuilder<?> builder)
      * @param key key
      * @param password password
      * @return keystore location as File
-     * @throws IOException
-     * @throws InterruptedException
+     * @throws IOException when opening or writing to file
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/clientproperties/AbstractKafkaClientProperties.java:259]: You should start with a capital letter.

File: operator-common/src/test/java/io/strimzi/operator/common/operator/MockCertManager.java [hunk 12]
```diff
@@ -315,7 +315,7 @@ public void generateSelfSignedCert(File keyFile, File certFile, Subject sbj, int
      * @param certFile path to the file which will contain the new self signed certificate
      * @param sbj      subject information
      * @param days     certificate duration
-     * @throws IOException
+     * @throws IOException  during writing to file
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/MockCertManager.java:318]: You should start with a capital letter. Maybe you can also add a verb?

```suggestion
     * @throws IOException  Thrown when writing to file fails
```

File: operator-common/src/test/java/io/strimzi/operator/common/operator/MockCertManager.java [hunk 13]
```diff
@@ -299,7 +299,7 @@ private static byte[] loadResource(InputStream is) {
      * @param certFile path to the file which will contain the self signed certificate
      * @param sbj      subject information
      * @param days     certificate duration
-     * @throws IOException
+     * @throws IOException  during writing to file
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/MockCertManager.java:302]: You should start with a capital letter. Maybe you can also add a verb?

```suggestion
     * @throws IOException  Thrown when writing to file fails
```

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java [hunk 14]
```diff
@@ -55,7 +55,7 @@ public void describeTo(final Description description) {
      * Checks all conditions in the supplied resource to see if the type of one of them matches the supplied rebalance state.
      *
      * @param state he expected rebalance state to be searched for.
-     * @return
+     * @return matcher for the List of Conditions
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java:58]: You should start with a capital letter.

**Review by scholzj — COMMENTED** [review]
Isn't it missing the `systemtests` module?

**Comment by im-konge** [comment]
> Isn't it missing the systemtests module?

That's covered by the `<!-- Skip Javadoc checks for tests-->` part in the suppressions.xml. I guess it doesn't need another rule now or?

**Comment by scholzj** [comment]
> > Isn't it missing the systemtests module?
> 
> That's covered by the `<!-- Skip Javadoc checks for tests-->` part in the suppressions.xml. I guess it doesn't need another rule now or?

I don't think it should ... what about issues like these: https://github.com/strimzi/strimzi-kafka-operator/blob/fd7427278eeeebc3ef1b008c0b9d4aa99434dc5f/systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java#L1102-L1122

**Comment by im-konge** [comment]
> I don't think it should ... what about issues like these: 

That doesn't seem to be a problem during the build. It has a problem with `@randomtag{}`, where the checkstyle expects that `{}` will contain some tag pointing to something else. But without the curly brackets, it seems to work.

**Comment by scholzj** [comment]
> > I don't think it should ... what about issues like these:
> 
> That doesn't seem to be a problem during the build. It has a problem with `@randomtag{}`, where the checkstyle expects that `{}` will contain some tag pointing to something else. But without the curly brackets, it seems to work.

Wait, you fixed those. I somehow missed it in the diff.

**Review by scholzj — APPROVED** [review]
Thanks.

**Pull request merged**